### PR TITLE
Task-8 redux+thunk setup

### PR DESCRIPTION
**Pull request by slolkunchik**
please check my setup of Redux.
I tested it on changing locale language event. If I'm right, the the best place for connect function is Root.js file. In Root component we implement our logic and it's the component, that know about the changing locale language event. 

@Viacheslav-ed  pay attention for structure of the project data. I assumed that the structure would be {en: dataEn, ru: dataRu, by: dataBy}

### CODE REVIEW

File: src/components/header/Header.jsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import styles from './header-style'
+import { Typography, IconButton } from '@material-ui/core'
+import { Select } from '@material-ui/core'
+import MenuIcon from '@material-ui/icons/Menu'
+import MenuItem from '@material-ui/core/MenuItem'
+
+const Header = ({handleOpenMenu, changeLocale, handleSelectChange, locale}) => {
+  const classes = styles()
+
+  const handleLocaleChange = ({ target: { value } }) => {
```
Comment (vvsvintsitsky) [inline src/components/header/Header.jsx]: It would be better to create this callback in the Root component.

File: src/components/HOC/MenuHOC.jsx [hunk 2]
```diff
@@ -5,19 +5,28 @@ export default navigations => {
   return Component => {
```
Comment (vvsvintsitsky) [inline src/components/HOC/MenuHOC.jsx]: HOCs are usually named like _withSomething_. Please, rename this file.

File: src/components/root.jsx [hunk 3]
```diff
@@ -1,14 +1,16 @@
 import React, { Fragment } from 'react'
 import { Switch, Route } from 'react-router-dom'
-import Header from './header'
-import Footer from '../components/Footer'
+import Header from './header/Header'
+// import Footer from '../components/Footer'
```
Comment (vvsvintsitsky) [inline src/components/root.jsx]: PR should not contain commented code.

File: src/actions/actionCreator.js [hunk 4]
```diff
@@ -0,0 +1,7 @@
+import { CHANGE_LOCALE } from '../constants'
```
Comment (vvsvintsitsky) [inline src/actions/actionCreator.js]: Action types, creators and reducer for locales should be created in one file.

File: src/components/header/Header.jsx [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+        </div>
+        <div className={classes.boxCenter}>
+          <div className={classes.logoText}>
+            <Typography variant="h4">Photographers of belarus</Typography>
+          </div>
+        </div>
+        <div className={classes.boxRight}>
+          <Select id="select" value={locale} onChange={handleLocaleChange}>
+            <MenuItem key='ruKey' value="Ru">Ru</MenuItem>
+            <MenuItem key='enKey' value="En">En</MenuItem>
```
Comment (vvsvintsitsky) [inline src/components/header/Header.jsx]: Hardcoded quantity of available languages.

File: src/components/root.jsx [hunk 6]
```diff
@@ -41,8 +51,33 @@ function Root({ handleOpenMenu }) {
...
+  locale: PropTypes.string.isRequired,
+}
+
+const mapStateToProps = store => {
+  return {
+    data: store.localeReducer
+  }
+}
+
+const mapDispatchToProps = dispatch => {
```
Comment (vvsvintsitsky) [inline src/components/root.jsx]: `const mapDispatchToProps  = { setLocaleAction: changeLocale };`

File: src/reducers/localeReducer.js [hunk 7]
```diff
@@ -0,0 +1,19 @@
+import { CHANGE_LOCALE } from '../constants'
+import RuData from '../data/dataRU-test'
+import EnData from '../data/data'
+
+const initialState = EnData;
+
+const localeReducer = (state = initialState, { locale, type }) => {
+  const newData = locale === 'Ru' ? RuData : EnData;
```
Comment (vvsvintsitsky) [inline src/reducers/localeReducer.js]: Hardcoded quantity of available languages.

File: src/actions/actionCreator.js (hunk 4 above)
Comment (slolkunchik) [reply src/actions/actionCreator.js]: in all examples I see, that action types constants, action creators and reducers are created in their own separate files. Is it bad practice?

**Comment by slolkunchik** [comment]
remarks are corrected (action types, creators and reducer are left for now).
connect in root is changed to hooks (just for my practice)

File: src/actions/actionCreator.js (hunk 4 above)
Comment (vvsvintsitsky) [reply src/actions/actionCreator.js]: They may be created in separate files, but in that case these files should be located in one folder (_src/locales/store/actionTypes.js_, _src/locales/store/actionCreators.js_, _src/locales/store/store.js_). For one file path would be _src/locales/store.js_.

Comment (slolkunchik) [reply src/actions/actionCreator.js]: @vvsvintsitsky ok, fixed

File: src/components/root.jsx [hunk 8]
```diff
@@ -23,26 +24,49 @@ const style = makeStyles(theme => ({
 function Root({ handleOpenMenu }) {
   const classes = style()
 
+  const [locale, setLocale] = useState( 'En')
```
Comment (vvsvintsitsky) [inline src/components/root.jsx]: This value should be stored in locales store.

**Pull request merged**